src/pages/CovidHistory.js

```diff
@@ -105,17 +114,38 @@ const CovidHistory = ({
                   max={moment().format(DATE_FORMAT.yyyymmdd)}
                   disabled={
                     `${history.field}` === "dateOfDose1Vaccination"
-                      ? !checkedOne
-                      : !checkedTwo
+                      ? !intakeState.covidVaccinationDose1Taken
+                      : `${history.field}` === "dateOfDose2Vaccination"
+                      ? !intakeState.covidVaccinationDose2Taken
+                      : `${history.field}` === "dateCovidBefore"
+                      ? !intakeState.covidPositiveEverBefore
+                      : null
+                  }
+                  value={
+                    `${history.field}` === "dateOfDose1Vaccination"
```

Comment:
please move all this stuff to a function, then you can set `value={getValue(history.field)}`